txs/envelope.go

```diff
@@ -109,7 +109,13 @@ func (txEnv *Envelope) Verify(chainID string) error {
 		return fmt.Errorf("%s: number of inputs (= %v) should equal number of signatories (= %v)",
 			errPrefix, len(inputs), len(txEnv.Signatories))
 	}
-	signBytes, err := txEnv.Tx.SignBytes()
+	var signBytes []byte
+	switch txEnv.GetEnc() {
+	case Envelope_RLP:
+		signBytes, err = txEnv.Tx.SignBytesRLPEncoded()
+	default:
+		signBytes, err = txEnv.Tx.SignBytes()
```

Comment:
Can we refactor this so we have `SignBytes(encodingType)` instead, I think it would be cleaner, particularly if we have ever have additional (i.e. protobuf) encoding types.